libvips/foreign/jxlsave.c

```diff
@@ -114,65 +120,227 @@ typedef struct _VipsForeignSaveJxl {
 	void *runner;
 	JxlEncoder *encoder;
 
-	/* The current y position in the frame, page height,
-	 * total number of pages, and the current page index.
-	 */
-	int write_y;
-	int page_height;
-	int page_count;
-	int page_number;
-
-	/* VipsRegion is not always contiguous, but we need contiguous RGB(A)
-	 * for libjxl. We need to copy each frame to a local buffer.
+	/* Buffer scanlines to make a line of libjxl tiles.
 	 */
-	VipsPel *frame_bytes;
-	size_t frame_size;
+	VipsPel *scanline_buffer;
+	size_t scanline_size;
+	int scanline_y;
 
 	/* Write buffer.
 	 */
 	uint8_t output_buffer[OUTPUT_BUFFER_SIZE];
 
+	/* Chunk reader.
+	 */
+	struct JxlChunkedFrameInputSource input_source;
+
+	/* Map thread ids to regions with this hash table, gate access to it with
+	 * the mutex.
+	 */
+	GHashTable *tile_hash;
+	GMutex tile_lock;
+
+	/* Current page we are saving.
+	 */
+	VipsImage *page;
+	VipsImage *page_colour;
+	VipsImage *page_extras;
```

Comment:
These two images seems to be unused.

Reply:
I think they were for >4 band support, which we've now dropped for now. I'll remove them.